lib/OpenQA/Worker.pm

```diff
@@ -104,6 +107,32 @@ sub catch_exit {
     }
 }
 
+sub setup_logging {
+    my ($app) = @_;
+
+    my $level = $worker_settings->{LOG_LEVEL} ? $worker_settings->{LOG_LEVEL} : "debug";
+
+    if (my $logfolder = $ENV{OPENQA_WORKER_LOG} || $worker_settings->{LOG_FOLDER}) {
+        # So each worker from each host get it's own log (as the folder can be shared). Hopefully the machine hostname
+        # is already sanitized. Otherwise we need to check
+        $app->log->path(catfile($logfolder, hostname() . "-$instance.log"));
+        $app->log->format(
+            sub {
+                my ($time, $level, @lines) = @_;
+                return '[' . localtime($time) . "] [" . $app->log_name . ":$level] " . join "\n", @lines, '';
+            });
+
+    }
+    elsif (lc($level) eq 'debug' || lc($level) eq 'info') {
```

Comment:
you should be safely able to it from here https://github.com/os-autoinst/openQA/pull/1448/files#diff-774870768a0b33acd9f09ccb5c364ba6R25